### DISCUSSION TITLE
[Feature] Set bulk geo-location only for photos that do not have location info

### DISCUSSION THREAD

**Discussion by anandv96** [immich-app/immich#12640] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

The ability to set a custom geo-location for only photos that do not already have any geo-location set already. 

This is useful for albums where some photos have geo-location data and others don't. One can set an approximate geo-location for the trip (which helps in looking at your trips on the map view), but retain the more precise GPS data for pictures that already had them. 

This way, you can add the geo-location to all the pictures on a trip to a particular city as the city centre, but not over-write the exact co-ordinates of the restaurant you had a great meal in.

Some people do not have the location enabled on their phones all the time, this could lead to some pictures with geo-location information and some without. This is also useful for people who use multiple cameras, some with GPS features and some without, or for shared albums.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
In the advanced search you can filter for "Country: Unknown" which will get you all assets without a location.

**Reply by anandv96** [reply]
Yes that's right. But that makes it a lot harder to select pictures by day.

It's a lot more convenient to mark a whole album or to mark multiple days at a time.

**Reply by Skivling** [reply]
It'd be nice as a quick action when many photos are selected, with option for all or only those with no gps data as you have already suggested.

**Comment by ovizii** [comment]
Can we set country only? At the moment it looks like I can only search for a city when editing the location of an image. I'd like to at least set the right country, even if I don't remember the exact location.

**Comment by bo0tzz** [comment]
There's now a location management utility that should cover this.

**Reply by ovizii** [reply]
> There's now a location management utility that should cover this.

There is but with very limited search/selection/exclusion options. See comments above.  
I hope that doesn't sound too negative, the tool is quite handy and a good starting point but lacks versatility.
